src/parser/parser_verification.js

```diff
@@ -0,0 +1,150 @@
+import { requiredValues } from '../util/requiredValues';
+import { parserUtils } from './parser_utils';
+
+/**
+ * Verify and trigger warnings if a node required value is not set.
+ * @param  {Node} node - The node element.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ * @param  {Boolean} [isAdInline] - True if node is contained inside a InLine node.
+ *
+ */
+function verifyRequiredValues(node, emit, isAdInline) {
+  if (!node || !node.nodeName) {
+    return;
+  }
+  if (node.nodeName === 'InLine') {
+    isAdInline = true;
+  }
+  verifyRequiredAttributes(node, emit);
+
+  if (hasSubElements(node)) {
+    verifyRequiredSubElements(node, emit, isAdInline);
+    for (const nodeKey in node.children) {
```

Comment:
I'm not sure but have you checked that `for...in` doesn't loop also through keys that you don't need. AFAIK `for...in` also loops from its constructor's prototype: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Reply:
I'll replace it with forEach

Reply:
Actually it's not an array it's an HTMLCollection so i can't use a forEach 😞 . I'll  replace for/in with the classical for loop.